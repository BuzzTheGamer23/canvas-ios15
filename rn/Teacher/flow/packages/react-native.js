// @flow
export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}
