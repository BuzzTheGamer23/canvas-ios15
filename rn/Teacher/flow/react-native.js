//
// Copyright (C) 2017-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow
/* eslint-disable */

// define basic style prop types
// the internal ones are the best we have
import {
  type ____ViewStyleProp_Internal,
  type ____TextStyleProp_Internal,
  type ____ImageStyleProp_Internal,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes'

declare type ViewStyleProp = ____ViewStyleProp_Internal
declare type TextStyleProp = ____TextStyleProp_Internal
declare type ImageStyleProp = ____ImageStyleProp_Internal
