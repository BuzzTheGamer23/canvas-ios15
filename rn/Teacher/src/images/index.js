// @flow

type Images = {
  course: {[name: string]: any},
  tabbar: {[name: string]: any},
  assignments: {[name: string]: any},
  rce: {[name: string]: any},
  [name: string]: any,
}

const images = {
  course: {
    announcements: require('./course/Announcements.png'),
    assignments: require('./course/Assignments.png'),
    discussions: require('./course/Discussions.png'),
    pages: require('./course/Pages.png'),
    people: require('./course/People.png'),
    quizzes: require('./course/Quiz.png'),
    syllabus: require('./course/Syllabus.png'),
    files: require('./course/Files.png'),
    settings: require('./course/Settings.png'),
    placeholder: require('./course/Placeholder.png'),
  },
  tabbar: {
    courses: require('./tabbar/courses.png'),
    inbox: require('./tabbar/inbox.png'),
    profile: require('./tabbar/profile.png'),
    staging: require('./tabbar/link.png'),
    stagingFilled: require('./tabbar/link-solid.png'),
  },
  assignments: {
    calendar: require('./assignments/Calendar.png'),
  },
  rce: {
    bold: require('./rce/bold.png'),
    embedImage: require('./rce/embed-image.png'),
    italic: require('./rce/italic.png'),
    link: require('./rce/link.png'),
    orderedList: require('./rce/ordered-list.png'),
    unorderedList: require('./rce/unordered-list.png'),
    redo: require('./rce/redo.png'),
    undo: require('./rce/undo.png'),
    active: {
      bold: require('./rce/bold-active.png'),
      italic: require('./rce/italic-active.png'),
      orderedList: require('./rce/ordered-list-active.png'),
      unorderedList: require('./rce/unordered-list-active.png'),
    },
  },
  speedGrader: {
    chatBubbleMe: require('./speedgrader/bubble-me.png'),
    chatBubbleThem: require('./speedgrader/bubble-them.png'),
    audio: require('./speedgrader/icon_audio.png'),
    video: require('./speedgrader/icon_video_clip.png'),
    page: require('./speedgrader/page.png'),
    pdf: require('./speedgrader/pdf-icon.png'),
    warning: require('./Warning.png'),
    longPress: require('./speedgrader/Longpress.png'),
    swipe: require('./speedgrader/Swipe.png'),
  },
  noTeacher: {
    parent: require('./Canvas-Parent.png'),
    student: require('./Canvas-Student.png'),
  },
  canvasLogo: require('./canvas-logo.png'),
  feedback: require('./feedback.png'),
  kabob: require('./kabob.png'),
  backIcon: require('./Back-icon.png'),
  check: require('./check-white.png'),
  starFilled: require('./star-filled.png'),
  starLined: require('./star-lined.png'),
  starLarge: require('./StarLarge.png'),
  add: require('./Add.png'),
  x: require('./x-icon.png'),
  clear: require('./Clear.png'),
  edit: require('./edit.png'),
  pencilBG: require('./Pencil-BG.jpg'),
  pickerArrow: require('./picker-arrow.png'),
  published: require('./Published.png'),
  unpublished: require('./Unpublished.png'),
  upArrow: require('./mini-arrow-up.png'),
  document: require('./Document.png'),
  attachment: require('./attachment.png'),
  mail: require('./Mail.png'),
  smallMail: require('./mail-small.png'),
  group: require('./group.png'),
  trash: require('./trash.png'),
  share: require('./share.png'),
}

export default (images: Images)
