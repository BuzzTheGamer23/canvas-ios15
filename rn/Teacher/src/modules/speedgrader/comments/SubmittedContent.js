// @flow

import React, { Component } from 'react'
import {
  View,
  StyleSheet,
  Image,
  TouchableOpacity,
} from 'react-native'
import colors from '../../../common/colors'
import { Title, SubTitle } from '../../../common/text'

export type SubmittedContentDataProps = {
  contentID: string,
  icon: any,
  title: string,
  subtitle: string,
}

export type SubmittedContentActionProps = {
  onPress: () => void,
}

type Props
  = SubmittedContentDataProps
  & SubmittedContentActionProps
  & {
    attachmentIndex: number,
    attemptIndex: number,
    submissionID: string,
  }

export default class SubmittedContent extends Component<any, Props, any> {

  selectContent = () => {
    this.props.onPress(this.props.submissionID, this.props.attemptIndex, this.props.attachmentIndex)
  }

  render () {
    return (
      <TouchableOpacity
        testID={`submitted-content.item-${this.props.contentID}`}
        style={styles.row}
        onPress={this.selectContent}
      >
        <Image
          testID={`submitted-content.icon-${this.props.contentID}`}
          style={styles.icon}
          source={this.props.icon}
        />
        <View style={styles.textContainer} >
          <Title
            testID={`submitted-content.title-${this.props.contentID}`}
            numberOfLines={1}
            ellipsizeMode='tail'
          >
            {this.props.title}
          </Title>
          <SubTitle
            testID={`submitted-content.subtitle-${this.props.contentID}`}
            numberOfLines={1}
            ellipsizeMode='tail'
          >
            {this.props.subtitle}
          </SubTitle>
        </View>
      </TouchableOpacity>
    )
  }
}

const styles = StyleSheet.create({
  row: {
    flexDirection: 'row',
    borderWidth: 1,
    borderColor: colors.seperatorColor,
    borderRadius: 4,
    overflow: 'hidden',
    paddingVertical: 6,
    paddingHorizontal: 8,
    marginBottom: 4,
    width: 304,
  },
  icon: {
    tintColor: colors.primaryButton,
  },
  textContainer: {
    flex: 1,
    marginHorizontal: 6,
  },
})
