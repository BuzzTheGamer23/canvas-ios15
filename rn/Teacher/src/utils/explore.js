/* @flow */

function hasProp (object: any, key: string, value: any): boolean {
  if (object instanceof Object) {
    const props = object.props
    if (props && props instanceof Object) {
      for (var prop in props) {
        if (prop === key && props[prop] === value) {
          return true
        }
      }
    }
  }

  return false
}

function selectByProp (object: any, propKey: string, value: any): ?any {
  if (!(typeof object === 'object')) {
    return null
  }

  if (hasProp(object, propKey, value)) {
    return object
  }

  let result = null

  for (var key in object) {
    if (object[key] instanceof Array) {
      object[key].forEach((e) => {
        result = selectByProp(e, propKey, value)
        if (result) {
          return result
        }
      })
    }

    result = selectByProp(object[key], propKey, value)
    if (result) {
      return result
    }
  }

  return result
}

class ComponentExplorer {
  component: any

  constructor (component: any) {
    this.component = component
  }

  selectByProp (propKey: string, value: any): ?any {
    return selectByProp(this.component, propKey, value)
  }

  selectByID (id: string): ?any {
    return this.selectByProp('testID', id)
  }
}

export default function explore (component: any): ComponentExplorer {
  return new ComponentExplorer(component)
}
