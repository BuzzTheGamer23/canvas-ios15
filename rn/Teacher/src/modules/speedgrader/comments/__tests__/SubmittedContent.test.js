// @flow

import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import SubmittedContent from '../SubmittedContent'
import images from '../../../../images'
import explore from '../../../../../test/helpers/explore'

jest.mock('TouchableOpacity', () => 'TouchableOpacity')

test('my chat bubbles render correctly', () => {
  const tree = renderer.create(
    <SubmittedContent
      contentID='1'
      icon={images.document}
      title='foo'
      subtitle='bar'
      onPress={jest.fn()}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('tapping calls onPress', () => {
  const onPress = jest.fn()
  const tree = renderer.create(
    <SubmittedContent
      contentID='1'
      icon={images.document}
      title='foo'
      subtitle='bar'
      onPress={onPress}
    />
  ).toJSON()
  const touchable = explore(tree).selectByID('submitted-content.item-1')
  touchable && touchable.props && touchable.props.onPress()
  expect(onPress).toHaveBeenCalled()
})
