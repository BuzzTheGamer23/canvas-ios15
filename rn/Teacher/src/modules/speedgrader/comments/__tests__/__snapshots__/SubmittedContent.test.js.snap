// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my chat bubbles render correctly 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    Object {
      "borderColor": "#C7CDD1",
      "borderRadius": 4,
      "borderWidth": 1,
      "flexDirection": "row",
      "marginVertical": 8,
      "maxWidth": 304,
      "overflow": "hidden",
      "paddingHorizontal": 8,
      "paddingVertical": 6,
    }
  }
  testID="submitted-content.item-1"
>
  <Image
    source={1}
    style={
      Object {
        "tintColor": "#374A59",
      }
    }
    testID="submitted-content.icon-1"
  />
  <View
    style={
      Object {
        "marginHorizontal": 6,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "fontFamily": ".SFUIDisplay",
          },
          Object {
            "color": "#2D3B45",
            "fontFamily": ".SFUIDisplay-semibold",
            "fontSize": 16,
          },
          undefined,
        ]
      }
      testID="submitted-content.title-1"
    >
      foo
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "fontFamily": ".SFUIDisplay",
          },
          Object {
            "color": "#7F91A7",
            "fontSize": 14,
          },
          undefined,
        ]
      }
      testID="submitted-content.subtitle-1"
    >
      bar
    </Text>
  </View>
</TouchableOpacity>
`;
